### PR TITLE
chore: disable access to grqphiql by default

### PR DESCRIPTION
**Pull request by mbacherycz**
Closes: [#1016](https://github.com/digdir/dialogporten-frontend/issues/1016)

### Dokumentasjon / Storybook / testdekning

- [ ] Dokumentasjon er oppdatert eller ikke relevant / nødvendig.
- [ ] Ny komponent har en eller flere `stories` i `Storybook`, eller så er ikke dette relevant.
- [ ] Det er blitt lagt til nye tester / eksiterende tester er blitt utvidet, eller tester er ikke relevant.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new environment variable, `ENABLE_GRAPHIQL`, allowing users to enable or disable the GraphiQL interface.
	- Conditional registration of the GraphiQL plugin based on the `enableGraphiql` configuration.
	- Enhanced Docker Compose configuration for the `bff` and `bff-migration` services with new environment variables for improved management.
	- Added a new parameter `graphiQLEnabled` to configuration files for flexible GraphiQL functionality control.

- **Bug Fixes**
	- Removed commented-out lines related to `ENABLE_GRAPHIQL`, clarifying its implementation status.

### CODE REVIEW

File: packages/bff/src/server.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ const startServer = async (): Promise<void> => {
   server.register(cookie);
 
   // Session setup
-  const { secret, enableHttps, cookieMaxAge } = config;
+  const { secret, enableHttps, cookieMaxAge, enable_graphiql } = config;
```
Comment (seanes) [inline packages/bff/src/server.ts:42]: Change to `enableGraphiQL` for consistent casing?

File: packages/bff/src/config.ts [hunk 2]
```diff
@@ -51,6 +52,7 @@ const config = {
   typeormSynchronizeEnabled: env.TYPEORM_SYNCHRONIZE_ENABLED,
   migrationRun: env.MIGRATION_RUN,
   dialogportenURL: env.DIALOGPORTEN_URL,
+  enable_graphiql: env.ENABLE_GRAPHIQL,
```
Comment (seanes) [inline packages/bff/src/config.ts:55]: Casing?

File: compose.yml [hunk 3]
```diff
@@ -108,6 +108,7 @@ services:
       HOSTNAME: http://app.localhost
       CLIENT_ID: ${CLIENT_ID}
       CLIENT_SECRET: ${CLIENT_SECRET}
+      ENABLE_GRAPHIQL: ${ENABLE_GRAPHIQL}
```
Comment (seanes) [inline compose.yml:111]: maybe just put it as `true` here? It's very useful for local development.

**Comment by seanes** [comment]
Should we enable this for dev / test, too, or rather change up the default to `true` instead, now at least with an opt-in to disable in prod? I find myself using it a lot tbh, but I don't think it has any public interest :D

**Comment by mbacherycz** [comment]
> Should we enable this for dev / test, too, or rather change up the default to `true` instead, now at least with an opt-in to disable in prod? I find myself using it a lot tbh, but I don't think it has any public interest :D

Make sense to me to have the console enabled in dev and disabled in prod.

File: .azure/applications/bff/main.bicep [hunk 4]
```diff
@@ -15,6 +15,7 @@ param dialogportenURL string
 param oicdUrl string
 param minReplicas int
 param maxReplicas int
+param graphiQLEnabled string = 'true'
```
[the remark this answers is not in the document]
Comment (arealmaas) [reply .azure/applications/bff/main.bicep:18]: Yeah, let's add this:

+@description('Controls whether GraphiQL interface is enabled. Should be disabled in production.') +@allowed([ + 'false' + 'true' +])

**Review by arealmaas — DISMISSED** [review]
LGTM! But would be nice to add the description and allowed directes above the parameter :)

File: .azure/applications/bff/main.bicep [hunk 5]
```diff
@@ -150,6 +151,10 @@ var containerAppEnvVars = [
   {
     name: 'NODE_ENV'
     value: 'production'
+  },
```
Comment (arealmaas) [inline .azure/applications/bff/main.bicep:154]: oh right, bicep is weird, we need to remove the `,`

**Review by arealmaas — APPROVED** [review]

**Review by seanes — APPROVED** [review]

**Pull request merged**